### DISCUSSION TITLE
Cromite AppImage

### DISCUSSION THREAD

**Discussion by CrazyAce25** [ivan-hc/AM#1011] [Q&A]
Ivan,

I recently wrote a go-appimage creation script using information I learned through your github page for Cromite in hopes that the developer would use said script to create an appimage release of Cromite.  When I did this, I posted a few more things regarding the possibilties that appimage, especially go-appimage and your AM/Appman provide, including sandboxing.  Well he came back and asked some questions about sandboxing that I cant answer because I can't get Appman to actually create the sandboxed image for me.  Since you have been at this for a while and are obviously proficient with your own software, I was hoping and would greatly appreciate if you could help me out by answering the developers questions regarding sandboxing with Appman.  The thread can be found here, https://github.com/uazo/cromite/issues/1579.   Thank you in advance!

**Comment by ivan-hc** [comment]
Hi, the developer is asking to show the result of `chrome://sandbox`, so you need to run the AppImage and add `chrome://sandbox` to the address bar of the browser.

About using it with AM/AppMan in sandbox, you need to perform the above test with or without the sandbox.

Have you got a ready-to-use AppImage?

**Comment by ivan-hc** [comment]
in brief, this is the test the developer is asking you (I use Chromium here)


https://github.com/user-attachments/assets/4fc43ab2-7c60-4ea6-a560-df63b60fcd6d

**Comment by ivan-hc** [comment]
no am sandbox
![no am sandbox](https://github.com/user-attachments/assets/54905707-ed8e-4de1-bd8a-d9d4e9c0e9e8)

with am sandbox
![with am sandbox](https://github.com/user-attachments/assets/0e0dc681-587a-4f0c-bc33-31cf917add1c)

in chromium the results does not changes

**Reply by CrazyAce25** [reply]
I understand how to run the test that he wants me to run, I just cant get AppMan to sandbox my Cromite AppImage.  

"./appman --sandbox cromite
ERROR: "cromite" is not installed"

**Reply by ivan-hc** [reply]
can I see the directory where you have installed the app?

**Comment by ivan-hc** [comment]
if you have not a repo or a fork from where you can install your AppImage using the option `-e` or `extra` (see https://github.com/ivan-hc/AM?tab=readme-ov-file#install-appimages-not-listed-in-this-database-but-available-in-other-github-repos) you can install `chromium` and replace the installed AppImage with yours renamed as `chromium`, in /opt/am/chromium or the related directory in AppMan if you use AppMan... and then perform my same test of the video

**Reply by CrazyAce25** [reply]
I dont have anywhere that my cromite is hosted to be able to install it via Appman.  Is that why Appman keeps giving me the error that cromite isnt installed, because it needs to be installed via AppMan in order to be sandboxed properly?  Also, you are welcome to download my build script from the link I supplied to the conversation above and bild the appimage real quick and then do the test with it if you are up for it.      

![Cromite](https://github.com/user-attachments/assets/c0fffd8c-4169-4e1a-b979-619f664745b3)

Thats is how Cromite looks before sandboxing but I cant get the after sandbox test done unfortunately.

**Comment by ivan-hc** [comment]
@CrazyAce25 do the following steps:
1. create the appimage and rename it as `chromium`
2. run `appman -i chromium aisap` to install both Chromium and Aisap (the latter is needed to sandboxes via bubblewrap... be sure bubblewrap is installed from your system package manager)
3. go in the directory where you have the apps installed with AppMan and remove the binary `chromium` from the directory named "chromium" (the directory where chromium is installed)
4. use your appimage of cromite that you have renamed as "chromium" at point 1 and put it into the directory "chromium"
5. perform the tests I've done in the video

**Reply by CrazyAce25** [reply]
I followed those steps.  I wasnt aware that chromium had to be installed by the package manager in order to be appimaged by AppMan.  So now that Ive got that part out of the way, Im now getting a new error.   

Error: "You need aisap for this script work

 ◆ DO YOU WISH TO INSTALL AISAP? Install size <5 MiB? (Y/n) "

I made sure it was installed via AppMan in case it needed to be in order to function but it just asks me over and over when I attempt to create the sandbox.

**Reply by ivan-hc** [reply]
is `aisap` installed?

**Reply by CrazyAce25** [reply]
Yes.  I installed it via Appman before ever running the command to sandbox and then the script wanted me to install it again so I said yes, but it still doesnt work.

**Reply by ivan-hc** [reply]
try to logoff or reboot

**Reply by CrazyAce25** [reply]
It was a path issue apparently.  "export PATH="$PATH:/home/oblive/.local/bin" got it working.

**Reply by CrazyAce25** [reply]
Thank you so much for the help, I really do appreciate it.  Hopefully this helps net us a Cromite appimage release.  On a side note, Im a big fan of your work as its helped me quite a bit with access to software and being able to avoid certain distros I  have no interest in ever using.  I was honestly thinking about learning a bit about using YAD so I could write a GUI front end for your AM/Appman package manager so that it gets more widespread use, as it really deserves more attention from the wider Linux community and a GUI always helps with that.

**Reply by ivan-hc** [reply]
thanks, a GUI would be a dream for me, but it's beyond my knowledge unfortunately.

**Comment by CrazyAce25** [comment]
![YAD](https://github.com/user-attachments/assets/4f912694-3f55-4e5f-8c32-060221954eb4)
![YAD1](https://github.com/user-attachments/assets/0fd8765b-aca6-4201-ac08-adb3a9ad126b)
![YAD2](https://github.com/user-attachments/assets/89946038-baaa-481e-a0f2-2661426e5097)
https://github.com/v1cont/yad
https://www.linux-magazine.com/Online/Blogs/Productivity-Sauce/Dress-Up-Bash-Scripts-with-YAD

Might be more doable than you think for someone with your obvious skill in bash scripting.

**Reply by ivan-hc** [reply]
I've actually used YAD before, and given up. I'm really bad at this stuff :disappointed_relieved:

**Reply by CrazyAce25** [reply]
Im still swamped with school, but if someone else doesnt build one in the meantime, then I will give it a shot at some point in the future.  YAD looked like the easiest option on its face, though Ive never used it before so I dont really know from first hand experience.  Ill continue to look more closely into the possible options with high ease of use to build a usable GUI frontend for your system.

**Reply by ivan-hc** [reply]
thank you

**Reply by CrazyAce25** [reply]
Out of curiosity, are you at all familiar with Python or interested in learning?  I ask because it would be orders of magnitude easier to take care of this GUI issue if the code base was ported to Python first.

**Reply by ivan-hc** [reply]
I am not familiar with python, in any way... in fact... sometimes I hate it, especially because there is never a single version to update, but multiple versions to work with, and as a result, an app will have to stick to older or newer versions, often being incompatible with each other. I see this every time I have to build an AppImage of an application written in python, and I always have problems building them.

On the other hand, someone in the past proposed a client in electron, based on my [catalog](https://portable-linux-apps.github.io/), all I had to do was provide them with a JSON file, [and I did](https://github.com/Portable-Linux-Apps/Portable-Linux-Apps.github.io/blob/main/apps.json)... but unfortunately the developer disappeared, so I stopped updating that file.

**Reply by Samueru-sama** [reply]
Python is a nightmare to make appimages of, look into Go or some other language that compiles statically instead.

**Reply by CrazyAce25** [reply]
Electron is rather resource heavy for what it accomplishes and tends to lack responsiveness, though I suppose it wouldnt be a bad option if you already had the javascript and node.js skills to make it happen, which I do not.
I definitely know what you mean with regards to the mess that Python has made of their ecosystem over the years. Backward compatibility isnt the best at all stages but mainly the issues between versions 2 and 3 as many things changed at that juncture.  Seeing as this venture would be starting out anew and not from an older python codebase there wouldnt be any of the afore mentioned issues to deal with.  There are at least 2 bash to python cli tools available on github to assit with the porting and combining those with the help of Code LLama-Python34B & Codestral Mamba22B LLMs, I dont think it would be too difficult to get your work ported over.  There are a plethora of options for GUIs within the Python world (https://realpython.com/learning-paths/python-gui-programming), and just having the codebase in Python increases the chances exponentially of a GUI being written for AM/Appman simply based on the number of developers there are within the eco-system.
 Just some things to think about and consider.

**Reply by CrazyAce25** [reply]
> Python is a nightmare to make appimages of, look into Go or some other language that compiles statically instead.

I was just looking at python due to its ease of use for scripting akin to bash, being as both are interpreted languages and come preinstalled on most linux installations.  It also has several options for GUI creation that make it attractive without being overly complicated.  Ive never looked into how well it functions with appimages, so this is the first Im hearing of these complications.  I have no experience with Go though unfortunately or any other compiled language for that matter.

**Reply by CrazyAce25** [reply]
> Python is a nightmare to make appimages of, look into Go or some other language that compiles statically instead.

Little did I know, but I am now starting to better understand where you were coming from with this and why.  https://harmful.cat-v.org/software/dynamic-linking/.  Im looking into a few statically compiled languages, Nim seemingly the most interesting of the bunch thus far.
